src/services/QueryService.ts

```diff
@@ -96,23 +95,31 @@ export class QueryService<T> {
   async get(add): Promise<BullhornListResponse<T>> {
     return this.run(add);
   }
-  async run(add): Promise<BullhornListResponse<T>> {
+  async run(add: boolean): Promise<BullhornListResponse<T>> {
     await this.initialized;
-    return Promise.all([this.http.get(this.endpoint, { params: this.parameters }), this.meta.getFull(this.parameters.fields, this.parameters.layout)])
-      .then(([response, metadata]) => [response.data, metadata])
-      .then(([result, metadata]: [BullhornListResponse<T>, BullhornMetaResponse]) => {
-        this._lastResponse = result;
-        const records = result.data;
-        if (add) {
-          this.records = this.records.concat(records);
-        } else {
-          this.records = records;
-        }
-        result.meta = metadata;
-        return result;
-      });
-  }
-  async then(done: any, fail?: any): Promise<any> {
+    const [response, metadata] = await Promise.all([this.httpGet(this.parameters), this.meta.getFull(this.parameters.fields, this.parameters.layout)]);
+    const result = response.data;
+
+    // TODO look at what was asked and what is return. And ask for more
+    const needMore = (this.parameters.count - result.count > 0) && this.parameters.fields.includes('billingCalendarInstances') && this.parameters.count === 500;
```

Comment:
We don't want to bake in a special case just for billingCalendarInstances. That just happens to be the only place that this currently occurs in the app, but we want this to apply anytime that more records are requested than are returned. We want to ask for more as long as there are more to ask for.  It's important to check that rest actually has more records, and can actually return them. See the java code at: https://github.com/bullhorn/dataloader/blob/master/src/main/java/com/bullhorn/dataloader/rest/RestApi.java#L240 for an example of exactly what you are wanting to do here.